### DISCUSSION TITLE
How to create a right fields for array of objects

### DISCUSSION THREAD

**Discussion by vadim-roshuor** [refinedev/refine#1883] [General]
Hello
I have a query:

![Screenshot from 2022-06-02 18-11-10](https://user-images.githubusercontent.com/102044742/171662119-26cd9a24-909b-47d5-8d9f-2ddcaf652a7b.png)

But I get an Error:

![Screenshot from 2022-06-02 18-12-39](https://user-images.githubusercontent.com/102044742/171662234-d2607f35-3e64-43ab-8679-aa8428adf6c5.png)

How to create a right fields for **image**. Image is array of objects

![Screenshot from 2022-06-02 18-17-24](https://user-images.githubusercontent.com/102044742/171662824-3bb6ebc6-5e02-4cff-8eb9-1d9f31cd9025.png)

**Comment by omeraplak** [comment]
Hey @vadim-roshuor ,
You can use nested fields. Like category field in our GraphQL doc;

https://refine.dev/docs/guides-and-concepts/data-provider/graphql/#list-page

also look graphql-query-builder package for writing more advanced queries,
https://github.com/atulmy/gql-query-builder

**Reply by vadim-roshuor** [reply]
thank you. you saved me. but I'm ashamed that I bothered you. everything is so simple. I don't even remember how I didn't notice it in the documentation

**Reply by omeraplak** [reply]
You're welcome. Please don't worry! 🎉 Don't forget to star us on GitHub and join [Discord](https://discord.gg/refine) 👊
